core/impl-base/src/main/java/org/jboss/arquillian/core/impl/ManagerImpl.java

```diff
@@ -597,6 +599,27 @@ void debug(ObserverMethod method, boolean interceptor)
       }
    }
 
+   private void debugExtension(Class<?> extension)
+   {
+      if (DEBUG)
+      {
+
+         if(eventStack == null)
+         {
+            eventStack = new ThreadLocal<Stack<Object>>()
+            {
+               @Override
+               protected Stack<Object> initialValue()
+               {
+                  return new Stack<Object>();
+               }
+            };
+         }
```

Comment:
I thought the same but if you check other parts of the class it is done in this way